src/rln/RLN.sol

```diff
@@ -54,158 +31,120 @@ contract RLN {
-        emit MemberWithdrawn(member.index);
-    }
+        require(_verifyProof(identityCommitment, proof), "RLN, withdraw: invalid proof");
 
-    /// @dev Releases stake amount.
-    /// @param identityCommitment: `identityCommitment` of withdrawn user.
-    function release(uint256 identityCommitment) external {
-        Withdrawal memory withdrawal = withdrawals[identityCommitment];
-        require(withdrawal.blockNumber != 0, "RLN, release: no such withdrawals");
-        require(block.number - withdrawal.blockNumber > FREEZE_PERIOD, "RLN, release: cannot release yet");
-
-        delete withdrawals[identityCommitment];
         delete members[identityCommitment];
-
-        token.safeTransfer(withdrawal.receiver, withdrawal.amount);
+        emit MemberExited(member.index);
     }
 
     /// @dev Slashes identity with identityCommitment.
     /// @param identityCommitment: `identityCommitment`;
-    /// @param receiver: stake receiver;
     /// @param proof: snarkjs's format generated proof (without public inputs) packed consequently.
-    function slash(uint256 identityCommitment, address receiver, uint256[8] calldata proof) external {
-        require(receiver != address(0), "RLN, slash: empty receiver address");
-
+    function slash(uint256 identityCommitment, uint256[8] calldata proof) external onlyRole(SLASHER_ROLE) {
         User memory member = members[identityCommitment];
         require(member.userAddress != address(0), "RLN, slash: member doesn't exist");
-        require(member.userAddress != receiver, "RLN, slash: self-slashing is prohibited");
-
-        require(_verifyProof(identityCommitment, receiver, proof), "RLN, slash: invalid proof");
+        require(_verifyProof(identityCommitment, proof), "RLN, slash: invalid proof");
 
+        //karma.slash(member.userAddress);
         delete members[identityCommitment];
-        delete withdrawals[identityCommitment];
 
-        uint256 withdrawAmount = member.messageLimit * MINIMAL_DEPOSIT;
-        uint256 feeAmount = (FEE_PERCENTAGE * withdrawAmount) / 100;
-
-        token.safeTransfer(receiver, withdrawAmount - feeAmount);
-        token.safeTransfer(FEE_RECEIVER, feeAmount);
-        emit MemberSlashed(member.index, receiver);
+        emit MemberSlashed(member.index, msg.sender);
     }
 
     /// @dev Groth16 proof verification
-    function _verifyProof(
-        uint256 identityCommitment,
-        address receiver,
-        uint256[8] calldata proof
-    )
-        internal
-        view
-        returns (bool)
-    {
+    function _verifyProof(uint256 identityCommitment, uint256[8] calldata proof) internal view returns (bool) {
         return verifier.verifyProof(
             [proof[0], proof[1]],
             [[proof[2], proof[3]], [proof[4], proof[5]]],
             [proof[6], proof[7]],
-            [identityCommitment, uint256(uint160(receiver))]
+            [identityCommitment, uint256(0)]
```

Comment:
@seugu Do you see any issues with doing it this way? We don't really have a receiver that will be able to withdraw the stake (as there's no actual stake)